### PR TITLE
master_CD-356939 Avoid using goroutines for Multipage Tiff2PNG Conver…

### PR DESCRIPTION
**Pull request by srajan-dadhich**
…sions

https://coupadev.atlassian.net/browse/CD-356939

- [ ] @revathi-murali 

- Multipage TIFF to PNG conversions was using Goroutines for processing each page. 
- But since now we only process a single page of the TIFF file, we don't need to spin-off multiple goroutines for processing different pages hence removing the goroutine implementation.

## Generated Reviewers

<details>
  <summary><em>:warning: Do not modify anything in this section! :warning:</em></summary>
  <em>The content in this section is managed automatically, and any manual changes to it will be ignored and overwritten the next time the code review status is refreshed! View <a href="https://cody.coupa.engineering/repos/coupa/bimg/pull/17">this PR</a> on Cody to see the current code review status.</em>
</details>

### CODE REVIEW

File: multipage_image.go [hunk 1]
```diff
@@ -8,65 +8,40 @@ import "C"
...
+	if err != nil {
+		tiffImage = &VIPSImage{image: out, err: err}
 	}
+	tiffImage = &VIPSImage{image: out, err: nil}
 
-	outVipsImage, err := vipsArrayJoin(frames)
-	if err != nil {
-		log.Errorf("[MultipageTIFFToPng] vipsArrayJoin err - %v", err)
-		return nil, err
+	var frame *C.VipsImage
```
Comment (revathi-murali) [inline multipage_image.go:28]: Minor: Do we need this?

File: multipage_image.go [hunk 2]
```diff
@@ -8,65 +8,40 @@ import "C"
...
+	defer C.g_object_unref(C.gpointer(frame))
 
-	outBuf, err := getImageBufferViaSlice(outVipsImage, PNG)
+	outBuf, err := getImageBufferViaSlice(frame, PNG)
 	if err != nil {
 		log.Errorf("[MultipageTIFFToPng] getImageBuffer err - %v", err)
 		return nil, err
 	}
+	if frame != nil {
+		C.g_object_unref(C.gpointer(frame))
```
Comment (revathi-murali) [inline multipage_image.go:44]: Do we need this as we already doing unref via defer in L36. Am I missing something here?

File: multipage_image.go [hunk 3]
```diff
@@ -8,65 +8,40 @@ import "C"
...
-	"sync"
 )
 
 type VIPSImage struct {
 	image *C.VipsImage
 	err   error
 }
 
-func MultipageTIFFToPng(buf []byte, pages int) ([]byte, error) {
+func MultipageTIFFToPng(buf []byte, page int) ([]byte, error) {
```
Comment (revathi-murali) [inline multipage_image.go:18]: Do you wanna change the method name as we longer do multipage tiff to png?

**Review by revathi-murali — APPROVED** [review]

File: multipage_image.go (hunk 3 above)
Comment (srajan-dadhich) [reply multipage_image.go:18]: Makes sense, I will update Imagica code then

File: multipage_image.go (hunk 2 above)
Comment (srajan-dadhich) [reply multipage_image.go:44]: yes, this is not required

File: multipage_image.go (hunk 1 above)
Comment (srajan-dadhich) [reply multipage_image.go:28]: yes, we don’t need this

File: multipage_image.go (hunk 3 above)
Comment (srajan-dadhich) [reply multipage_image.go:18]: @revathi-murali in this function we are using methods from package bimg like `vipsTIFFReadWithAlpha` and `getImageBufferViaSlice` which are not exported, so we won't be able to use them in Imagica, so let's keep this function here in bimg itself

**Review by revathi-murali — APPROVED** [review]

**Comment by srajan-dadhich** [comment]
[lockit v5.0.2] merged via Lockit CLI on 2023-05-22 15:38:51.511756 +0530 IST m=+6.861478264

**Pull request merged**